consensus/qbft/src/main/java/org/hyperledger/besu/consensus/qbft/messagewrappers/Proposal.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.consensus.qbft.messagewrappers;
+
+import org.hyperledger.besu.consensus.common.bft.messages.SignedData;
+import org.hyperledger.besu.consensus.qbft.payload.PayloadSerializers;
+import org.hyperledger.besu.consensus.qbft.payload.PreparePayload;
+import org.hyperledger.besu.consensus.qbft.payload.ProposalPayload;
+import org.hyperledger.besu.consensus.qbft.payload.RoundChangePayload;
+import org.hyperledger.besu.ethereum.rlp.BytesValueRLPOutput;
+import org.hyperledger.besu.ethereum.rlp.RLP;
+import org.hyperledger.besu.ethereum.rlp.RLPInput;
+
+import java.util.List;
+
+import org.apache.tuweni.bytes.Bytes;
+
+public class Proposal extends IbftMessage<ProposalPayload> {
+
+  private final List<SignedData<RoundChangePayload>> roundChanges;
+  private final List<SignedData<PreparePayload>> prepares;
+
+  public Proposal(
+      final SignedData<ProposalPayload> payload,
+      final List<SignedData<RoundChangePayload>> roundChanges,
+      final List<SignedData<PreparePayload>> prepares) {
+    super(payload);
+    this.roundChanges = roundChanges;
+    this.prepares = prepares;
+  }
+
+  public List<SignedData<RoundChangePayload>> getRoundChanges() {
+    return roundChanges;
+  }
+
+  public List<SignedData<PreparePayload>> getPrepares() {
+    return prepares;
+  }
+
+  @Override
+  public Bytes encode() {
+    final BytesValueRLPOutput rlpOut = new BytesValueRLPOutput();
+    rlpOut.startList();
+    getSignedPayload().writeTo(rlpOut);
+    rlpOut.writeList(roundChanges, SignedData::writeTo);
+    rlpOut.writeList(roundChanges, SignedData::writeTo);
+    rlpOut.endList();
+    return rlpOut.encoded();
+  }
+
+  public static Proposal decode(final Bytes data) {
+    final RLPInput rlpIn = RLP.input(data);
+    rlpIn.enterList();
+    final SignedData<ProposalPayload> payload =
+        PayloadSerializers.readSignedProposalPayloadFrom(rlpIn);
+    final List<SignedData<PreparePayload>> prepares =
```

Comment:
According to the spec the prepares should be the last field included, not the 2nd field.

Reply:
done